devbox.go

```diff
@@ -313,17 +313,50 @@ func missingDevboxJSONError(dir string) error {
 			}
 		}
 	}
-	return usererr.New("No devbox.json found in %s, or any parent directories. Did you run `devbox init` yet?", dir)
+
+	parentDirCheckAddendum := ""
+	if didCheckParents {
+		parentDirCheckAddendum = ", or any parent directories"
+	}
+
+	return usererr.New("No devbox.json found in %s%s. Did you run `devbox init` yet?", dir, parentDirCheckAddendum)
 }
 
+// findConfigDir
 func findConfigDir(dir string) (string, error) {
 
 	// Sanitize the directory and use the absolute path as canonical form
-	cur, err := filepath.Abs(dir)
+	absDir, err := filepath.Abs(dir)
 	if err != nil {
 		return "", errors.WithStack(err)
 	}
 
+	// If the directory (`dir`) is specified, then we check just that directory.
+	// Otherwise, we search the parent directories.
+	if dir != "" {
+		fi, err := os.Stat(dir)
+		if err != nil {
+			return "", err
+		}
+		switch mode := fi.Mode(); {
+		case mode.IsDir():
+			if !plansdk.FileExists(filepath.Join(absDir, configFilename)) {
+				return "", missingDevboxJSONError(dir, false /*didCheckParents*/)
```

Comment:
well, it won't quite be:
> return "", &missingConfigError{dir: dir, checkedParents: false}

because we need to run some logic in `missingDevboxJSONError` to make the error message read nicely.

Reply:
🤔  if the struct implements the `Error` interface with the same logic in `missingDevboxJSONError`, would that work?

Reply:
That's what I was thinking. Just make the `Error` method format the message however we want.